iosxr_setup.py

```diff
@@ -113,60 +141,58 @@ def setup_networking(self, host_ip=None, gateway=None, allow_root_login=False):
 
         # Needed for jenkins if using root password
         if allow_root_login:
-            xr1.send("run sed -i 's/PermitRootLogin no/PermitRootLogin yes/' /etc/ssh/sshd_config_operns")
-
-        #
-        # Send commands to XR Linux
-        #
-        xr1.send("run")
+            xr1.send("bash -c sed -i 's/PermitRootLogin no/PermitRootLogin yes/' /etc/ssh/sshd_config_operns")
 
         # Add passwordless sudo as required by jenkins
-        xr1.send("echo '####Added by iosxr_setup to give vagrant passwordless access' >> /etc/sudoers")
-        xr1.send("echo '%sudo ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers")
+        # sudo not vagrant because we are operating in xrnns and global-vrf user space
```

Comment:
BTW the echo to sudoers - you mentioned echo issues.

you tried a subshell ?

sudo sh -c "echo \"group ALL=(user) NOPASSWD: ALL\" >> /etc/sudoers"

sunstone.sh does both these approaches
        su -c "cat <<EOF >> /etc/sudoers
$LOGNAME ALL=(ALL:ALL) ALL
$LOGNAME ALL=(ALL) NOPASSWD:ALL
EOF"

        echo $LOGNAME ALL=NOPASSWD: ALL | sudo tee -a /etc/sudoers



On 1 July 2016 at 11:11, Glenn Matthews <notifications@github.com> wrote:

> In iosxr_setup.py
> <https://github.com/ios-xr/iosxrv-x64-vbox/pull/3#discussion_r69314009>:
>
> >
> >          # Add passwordless sudo as required by jenkins
> > -        xr1.send("echo '####Added by iosxr_setup to give vagrant passwordless access' >> /etc/sudoers")
> > -        xr1.send("echo '%sudo ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers")
> > +        # sudo not vagrant because we are operating in xrnns and global-vrf user space
>
> Not sure I understand this comment or the change back from vagrant to
> %sudo - can you explain?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ios-xr/iosxrv-x64-vbox/pull/3/files/66cffbc15a8cb663494ffdaf3dd06ca6af25e05e#r69314009>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAYNnw0sfV6hlYxwvaVXuBLx-noJMWsaks5qRS4SgaJpZM4JCqR5>
> .
>